core/lib/zksync_core/src/l1_gas_price/singleton.rs

```diff
@@ -59,17 +49,23 @@ impl GasAdjusterSingleton {
                 .context("GasAdjuster::new()")?;
                 Ok(Arc::new(adjuster))
             })
-            .await;
-        adjuster.clone()
+            .await
+        {
+            Ok(adjuster) => Ok(adjuster.clone()),
+            Err(_e) => Err(anyhow::anyhow!("Failed to get or initialize GasAdjuster")),
+        }
     }
 
     pub fn run_if_initialized(
         self,
         stop_signal: watch::Receiver<bool>,
     ) -> Option<JoinHandle<anyhow::Result<()>>> {
-        let gas_adjuster = self.singleton.get()?.clone();
+        let gas_adjuster = match self.singleton.get()? {
+            Ok(gas_adjuster) => gas_adjuster.clone(),
+            Err(_e) => return None,
+        };
```

Comment:
Nit: I guess you can do it as `self.singleton.get()?.ok()?.clone()`.

Reply:
Similar to `get_or_init()`, the `get()` method result can't be dereferenced because the `anyhow::Error` is not cloneable.